### ISSUE TITLE
v.4.x: AssertionFailedException in Decompiler.IL.Branch.CheckInvariant

### ISSUE THREAD

**Issue by greenozon** [icsharpcode/ILSpy#1351]
ILSpy version 4.0.0.4467-beta3

```
Error decompiling Microsoft.CodeAnalysis.VisualBasic.DocumentationCommentCrefBinder.CreateTypeParameterSymbolsAndConstructSymbols
 ---> ICSharpCode.ILSpy.AssertionFailedException: 
   at ICSharpCode.ILSpy.ILSpyTraceListener.Fail(String message, String detailMessage) in offset 380
   at ICSharpCode.ILSpy.ILSpyTraceListener.Fail(String message) in offset 9
   at System.Diagnostics.TraceInternal.Fail(String message) in offset 55
   at System.Diagnostics.Debug.Assert(Boolean condition) in offset 6
   at ICSharpCode.Decompiler.IL.Branch.CheckInvariant(ILPhase phase) in offset 65
   at ICSharpCode.Decompiler.IL.ILInstruction.CheckInvariant(ILPhase phase) in offset 126
   at ICSharpCode.Decompiler.IL.Block.CheckInvariant(ILPhase phase) in offset 8
   at ICSharpCode.Decompiler.IL.Block.RunTransforms(IEnumerable`1 transforms, BlockTransformContext context) in offset 77
   at ICSharpCode.Decompiler.IL.Transforms.BlockILTransform.VisitBlock(ControlFlowNode cfgNode, BlockTransformContext context) in offset 146
   at ICSharpCode.Decompiler.IL.Transforms.BlockILTransform.VisitBlock(ControlFlowNode cfgNode, BlockTransformContext context) in offset 89
   at ICSharpCode.Decompiler.IL.Transforms.BlockILTransform.VisitBlock(ControlFlowNode cfgNode, BlockTransformContext context) in offset 89
   at ICSharpCode.Decompiler.IL.Transforms.BlockILTransform.VisitBlock(ControlFlowNode cfgNode, BlockTransformContext context) in offset 89
   at ICSharpCode.Decompiler.IL.Transforms.BlockILTransform.VisitBlock(ControlFlowNode cfgNode, BlockTransformContext context) in offset 89
   at ICSharpCode.Decompiler.IL.Transforms.BlockILTransform.VisitBlock(ControlFlowNode cfgNode, BlockTransformContext context) in offset 89
   at ICSharpCode.Decompiler.IL.Transforms.BlockILTransform.Run(ILFunction function, ILTransformContext context) in offset 146
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.DecompileBody(IMethod method, EntityDeclaration entityDecl, DecompileRun decompileRun, ITypeResolveContext decompilationContext) in offset 581
-- continuing with outer exception (ICSharpCode.Decompiler.DecompilerException) --
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.DecompileBody(IMethod method, EntityDeclaration entityDecl, DecompileRun decompileRun, ITypeResolveContext decompilationContext) in offset 1068
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.DoDecompile(IMethod method, DecompileRun decompileRun, ITypeResolveContext decompilationContext) in offset 249
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.Decompile(IList`1 definitions) in offset 408
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.Decompile(EntityHandle[] definitions) in offset 1
   at ICSharpCode.ILSpy.CSharpLanguage.DecompileMethod(IMethod method, ITextOutput output, DecompilationOptions options) in offset 246
   at ICSharpCode.ILSpy.TreeNodes.MethodTreeNode.Decompile(Language language, ITextOutput output, DecompilationOptions options) in offset 15
   at ICSharpCode.ILSpy.TextView.DecompilerTextView.DecompileNodes(DecompilationContext context, ITextOutput textOutput) in offset 69
   at ICSharpCode.ILSpy.TextView.DecompilerTextView.<>c__DisplayClass34_0.<DecompileAsync>b__0() in offset 38

```

target:  https://www.sendspace.com/file/m0wto2

**Comment by siegfriedpammer** [comment]
I cannot reproduce this in the latest version.

**Comment by greenozon** [comment]
You are right!

when I've put just this one dll into separate dir, the is no assert, instead I see some comments like this:

```
// Microsoft.CodeAnalysis.VisualBasic.DocumentationCommentCrefBinder
using Microsoft.CodeAnalysis.VisualBasic.Symbols;
using Microsoft.CodeAnalysis.VisualBasic.Syntax;
using System.Collections.Generic;

private static void CreateTypeParameterSymbolsAndConstructSymbols(GenericNameSyntax genericName, ArrayBuilder<Symbol> symbols, Dictionary<string, CrefTypeParameterSymbol> typeParameters)
{
	//IL_0006: Unknown result type (might be due to invalid IL or missing references)
	//IL_000b: Unknown result type (might be due to invalid IL or missing references)
	//IL_0056: Unknown result type (might be due to invalid IL or missing references)
	//IL_005b: Unknown result type (might be due to invalid IL or missing references)
	//IL_00bd: Unknown result type (might be due to invalid IL or missing references)
	//IL_00c2: Unknown result type (might be due to invalid IL or missing references)
	//IL_00c4: Unknown result type (might be due to invalid IL or missing references)
	//IL_00c7: Invalid comparison between Unknown and I4
	//IL_00c9: Unknown result type (might be due to invalid IL or missing references)
	//IL_00cd: Unknown result type (might be due to invalid IL or missing references)
	//IL_00d0: Invalid comparison between Unknown and I4
	//IL_00d4: Unknown result type (might be due to invalid IL or missing references)
	//IL_00d8: Invalid comparison between Unknown and I4
	//IL_00da: Unknown result type (might be due to invalid IL or missing references)
	//IL_00de: Invalid comparison between Unknown and I4
```


I'll need to identify the full set of ref-ed dlls to reproduce the issue

**Comment by greenozon** [comment]
Found missed dll that lead to the issue,
pls drop this nearby the one from 1st post

https://www.sendspace.com/file/to8lxi

and you'll have a chance to see the assert  (I hope) :)

BTW, used this time this ver:  ILSpy version 4.0.0.4468-beta3

**Comment by siegfriedpammer** [comment]
Fixed in 29527b8.
